miner/src/miner.rs

```diff
@@ -175,6 +176,51 @@ impl MinerService for Miner {
 		}
 	}
 
+	fn call(&self, t: &SignedTransaction, chain: &BlockChainClient) -> Result<Executed, Error> {
+		let sealing_work = self.sealing_work.lock().unwrap();
+		let block = try!(sealing_work.peek_last_ref().ok_or(ExecutionError::Internal)).block();
+		let header = block.header();
+		let last_hashes = chain.last_hashes();
+		let env_info = EnvInfo {
+			number: header.number(),
+			author: header.author().clone(),
+			timestamp: header.timestamp(),
+			difficulty: header.difficulty().clone(),
+			last_hashes: last_hashes,
+			gas_used: U256::zero(),
+			gas_limit: U256::max_value(),
+		};
+		// that's just a copy of the state.
+		let mut state = block.state().clone();
+		let sender = try!(t.sender());
+		let balance = state.balance(&sender);
+		// give the sender max balance
+		state.sub_balance(&sender, &balance);
+		state.add_balance(&sender, &U256::max_value());
+		let options = TransactOptions { tracing: false, check_nonce: false };
+		Executive::new(&mut state, &env_info, chain.engine()).transact(t, options)
+	}
+
+	fn balance(&self, address: &Address) -> U256 {
+		let sealing_work = self.sealing_work.lock().unwrap();
+		sealing_work.peek_last_ref().map_or(x!(0), |b| b.block().fields().state.balance(address))
```

Comment:
what about when there's no work (`peer_last_ref()` returns `None`) and it is querying an account with a non-zero balance?